sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/AutoScaler.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.jms;
+
+import java.io.Serializable;
+
+/**
+ * An interface allowing the implementation of getBacklogBytes and getSplitBacklogBytes.
+ *
+ * <p>The interface also provides the following methods: - start to start the autoscaler - stop to
+ * stop the autoscaler
+ */
+public interface AutoScaler extends Serializable {
+
+  void start();
+
+  long getTotalBacklogBytes();
+
+  long getSplitBacklogBytes();
```

Comment:
I think we should drop this method since we have no way to provide to the scaler with what `split` we are currently processing since in the future it could change over time.
```suggestion
```